### PR TITLE
Add massive price update tool and some improvements

### PR DESCRIPTION
**Pull request by IC-Ilies**

### CODE REVIEW

File: tools/price_impact.php [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+}
+else{
+	$token = newToken();
+}
+
+
+if($action == 'confirm_massive_update'){
+
+	if(!empty($updateproductprices)) {
+		$sql = "SELECT rowid FROM ".MAIN_DB_PREFIX."product WHERE ref LIKE '".$ref."%'";
```
Comment (atm-john) [inline tools/price_impact.php:35]: Warning : SQL injection possible

Comment (IC-Ilies) [reply tools/price_impact.php]: Thank you ! I commited a fix, ref and temp_variation (even if not necessary ?) are escaped ;)

File: tools/replace_url.php [hunk 2]
```diff
@@ -126,31 +126,40 @@ function __processUrlReplace($oldUrl, $newUrl, $logManager, $replaceDomaineOnly
 
 
 	$logManager->addLog($langs->trans('ReplaceUrlXByY', $oldUrl, $newUrl));
+	$db->begin();
+	$tables = array('c_email_templates' => array('content'), 'user' => array('signature'), 'mailing' => array('sujet', 'body'));
+
+	if ($use_rollback) {
+		$db->begin();
+	}
```
Comment (atm-john) [inline tools/replace_url.php:134]:
```suggestion
```

Already add 2 lines before

File: tools/replace_url.php [hunk 3]
```diff
@@ -126,31 +126,40 @@ function __processUrlReplace($oldUrl, $newUrl, $logManager, $replaceDomaineOnly
...
-		}
+	}
+
+	if (!empty($logManager->getErrors()) && $use_rollback) {
+		$db->rollback();
+		$logManager->addError($langs->trans("UsedRollback"));
+	} else {
+		$db->commit();
+		$logManager->addError($langs->trans("Commited"));
 	}
```
Comment (atm-john) [inline tools/replace_url.php:164]: Your error management is broken

**Pull request merged**